diff --git a/Source/Core/Common/FileUtil.cpp b/Source/Core/Common/FileUtil.cpp
--- a/Source/Core/Common/FileUtil.cpp
+++ b/Source/Core/Common/FileUtil.cpp
@@ -45,6 +45,7 @@
 #endif
 
 #if defined(__APPLE__)
+#include <TargetConditionals.h>
 #include <CoreFoundation/CFBundle.h>
 #include <CoreFoundation/CFString.h>
 #include <CoreFoundation/CFURL.h>
@@ -766,7 +767,7 @@ static std::string CreateSysDirectoryPath()
 {
-#if defined(_WIN32) || defined(LINUX_LOCAL_DEV)
+#if defined(_WIN32) || (defined(LINUX_LOCAL_DEV) && !defined(__APPLE__))
 #define SYSDATA_DIR "Sys"
-#elif defined(__APPLE__) && defined(__IPHONE_OS_VERSION_MIN_REQUIRED)
+#elif defined(__APPLE__) && TARGET_OS_IPHONE
 #define SYSDATA_DIR "Sys"
 #elif defined __APPLE__
 #define SYSDATA_DIR "Contents/Resources/Sys"
